community/community-it/bolt-it/src/test/java/org/neo4j/bolt/transport/RejectTransportEncryptionIT.java

```diff
@@ -95,11 +82,16 @@ public void teardown() throws Exception
         }
     }
 
-    @Test
-    public void shouldRejectConnectionAfterHandshake() throws Throwable
+    @ParameterizedTest( name = "{displayName} {index}" )
+    @MethodSource( "transportFactory" )
+    public void shouldRejectConnectionAfterHandshake( Factory<TransportConnection> cf, Exception expected )
     {
-        exception.expect( expected.getClass() );
-        exception.expectMessage( expected.getMessage() );
-        client.connect( server.lookupDefaultConnector() ).send( util.defaultAcceptedVersions() );
+        this.client = cf.newInstance();
+
+        assertThrows( expected.getClass(), () ->
+        {
+            client.connect( server.lookupDefaultConnector() ).send( util.defaultAcceptedVersions() );
+
+        }, expected.getMessage());
```

Comment:
This does not actually check the message, that parameter is the error message if the assert fails. You have to do it like this:

```suggestion
        var exception = assertThrows( expected.getClass(), () -> client.connect( server.lookupDefaultConnector() ).send( util.defaultAcceptedVersions() ),
                                      expected.getMessage() );
        assertEquals( expected.getMessage(), exception.getMessage() );
```

Reply:
Thanks for hinting.

I also removed the Java 9 condition on the same file.